libarchive/archive_read_support_format_tar.c

```diff
@@ -631,8 +631,6 @@ archive_read_format_tar_read_data(struct archive_read *a,
 		}
 
 		*buff = __archive_read_ahead(a, 1, &bytes_read);
-		if (bytes_read < 0)
```

Comment:
Whenever `__archive_read_ahead` puts an error value into `bytes_read` it also returns NULL, so the `bytes_read < 0` checks throughout are redundant and have been removed.